### DISCUSSION TITLE
IP Ping and discover automatically

### DISCUSSION THREAD

**Discussion by Wangjxg** [netbox-community/netbox#9618] [General]
In 'IP Addresses', is there any other way to judge whether IP is active and discover other active ip automatically ?

**Comment by candlerb** [comment]
Netbox is not a network discovery system, nor is it a monitoring portal.  It's a configuration database or "source of truth". In other words, you use Netbox to record what IP addresses *should* be present on your network, and other tools can safely configure themselves from that known-good data.

If you want to discover other devices which have been plugged into your network without your knowledge, then you'll need a separate tool for this (e.g. try Netdisco).  Then it will be up to you how to resolve the discrepancy: maybe you need to unplug the rogue device, or maybe you add the device into Netbox as valid and required (but was forgotten).

Netbox can't decide for you whether the device *should* be present or not on your network.

**Reply by Wangjxg** [reply]
Thanks for your reply

**Reply by emersonfelipesp** [reply]
**Netbox supports plugins, which allows you to develop your own integrations with third-party systems.**

Although the main purpose of Netbox is really to be the Source of Truth of the network as @candlerb said, you can read about **[Netbox Plugins Development](https://docs.netbox.dev/en/stable/plugins/development/)**, how API works and the Django framework. With this, it's pretty possible to do what you want and solve your "problem"!
